var/spack/repos/builtin/packages/root/package.py

```diff
@@ -31,71 +43,431 @@ class Root(CMakePackage):
         patch('math_uint.patch', when='@6.06.02')
         patch('root6-60606-mathmore.patch', when='@6.06.06')
 
-    variant('graphviz', default=False, description='Enable graphviz support')
+    # ###################### Variants ##########################
+
+    variant('avahi', default=False,
+        description='Compile with avahi')
+    variant('aqua', default=False,
+        description='Enable Aqua interface')
+    # No need for a specific variant: libafterimage is not provided by spack
+    # By default always true, we get the builtin included in the source
+    # variant('asimage', default=True,
+    #    description='Enable image processing support')
+    variant('davix', default=True,
+        description='Compile with external Davix')
+    variant('emacs', default=False,
+        description='Enable Emacs support')
+    variant('examples', default=True,
+        description='Install examples')
+    variant('fftw', default=False,
+        description='Enable Fast Fourier Transform support')
+    variant('fits', default=False,
+        description='Enable support for images and data from FITS files')
+    variant('fortran', default=False,
+        description='Enable the Fortran components of ROOT')
+    variant('graphviz', default=False,
+        description='Enable graphviz support')
+    variant('gdml', default=True,
+        description='Enable GDML writer and reader')
+    variant('gsl', default=True,
+        description='Enable linking against shared libraries for GSL')
+    variant('http', default=False,
+        description='Enable HTTP server support')
+    variant('jemalloc', default=False,
+        description='Enable using the jemalloc allocator')
+    variant('kerberos', default=False,
+        description='Enable Kerberos support')
+    variant('ldap', default=False,
+        description='Enable LDAP support')
+    variant('libcxx', default=False,
+        description='Build using libc++')
+    variant('math', default=True,
```

Comment:
Unlike most variants, this does not appear to activate a dependency. Is it expensive to build (or does it fail on some systems)?

It's fine to do this, but FWIW it can reduce complexity to just build optional things unconditionally.

Reply:
I agree it does not activate any dependency, however it triggers other build options that otherwise should not enabled, so I would prefer to keep.